tests/tests/integration_tests.rs

```diff
@@ -456,21 +465,22 @@ async fn ws_server_subscribe_with_stream() {
 	let mut sub2: Subscription<usize> =
 		client.subscribe("subscribe_10_ints", None, "unsubscribe_10_ints").await.unwrap();
 	tracing::info!("[test] Subscribed");
-	let r = sub1.next().await;
-	tracing::debug!("[test] sub1 next: {r:?}");
-	let r = sub2.next().await;
-	tracing::debug!("[test] sub2 next: {r:?}");
-	let r = sub1.next().await;
-	tracing::debug!("[test] sub1 next: {r:?}");
-	let r = sub1.next().await;
-	let r = sub1.next().await;
-	tracing::debug!("[test] sub1 next: {r:?}");
-	let r = sub2.next().await;
-	tracing::debug!("[test] sub2 next: {r:?}");
+
+	assert_eq!(sub1.next().await.unwrap().unwrap(), 1);
+	assert_eq!(sub2.next().await.unwrap().unwrap(), 1);
+	assert_eq!(sub1.next().await.unwrap().unwrap(), 2);
+	assert_eq!(sub2.next().await.unwrap().unwrap(), 2);
+	assert_eq!(sub2.next().await.unwrap().unwrap(), 3);
+
+	// Be rude, don't run the destructor
+	std::mem::forget(sub2);
+	// Sub1 is still in business
+	assert_eq!(sub1.next().await.unwrap().unwrap(), 3);
+
 	// terminate connection.
+	// TODO: (dp) removing the drop changes nothing. Why is that?
```

Comment:
because some messages might already have been received and "buffered" so you will have to read the 10 items to be on the safe-side.

depending on execution time

Reply:
I'm still confused. I understand what you're saying here and it makes sense. The client has already received the items from the server, just that my test code hasn't read all of them yet. But I don't understand what this last assert actually does. We should be getting 10 `()`s sent on `rx` (and we do; if I read one more value the test hangs ofc) but how is that tied to dropping the client? It seems to act the same with and without dropping it?

Reply:
the client may or may have received the items we can't know for user without a mutex or sending them over the channel.

"the last assert" below asserts ensure that server terminated the "sub" after the "the subscription was closed that".
For every subscription that the server closes a message is sent on this channel.
The entire point of that channel is to indicate that the subscription(s) was closed by the server, thus I think don't think it's necessary in this test because we have another test for that.

Thus, if you want that know that you should read the channel twice.